src/main/scala/scalaz/parsers/Parsing.scala

```diff
@@ -265,6 +265,11 @@ sealed trait Parsing[F[_], G[_], E] {
     def pure[I, A](a: A): Codec[I, A] =
       apply(Equiv.lift[I, (I, A)]((_, a), _._1))
 
+    def delay[I, A](pa: => Codec[I, A]): Codec[I, A] = {
+      lazy val pa0 = pa
```

Comment:
Before reviewing further I am curious if you have seen any articles wrt drawbacks to using lazy val in scala:  https://blog.codecentric.de/en/2016/02/lazy-vals-scala-look-hood/

Reply:
Yes, I saw such articles, thanks.